### ISSUE TITLE
BUG: Incomplete 'Settings' Raises Errors the Next Time the Program Runs

### ISSUE THREAD

**Issue by huizm** [BiDuang/ModSyncer#2]
I interrupted the program on the fly using `^C` (or `Ctrl+C` in Windows) when testing a new feature, thus leading to an error the next time it ran as below.
```
Traceback (most recent call last):
  File "~/Desktop/ModSyncer/modsyncer.py", line 59, in <module>
    settings = json.load(f)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/json/__init__.py", line 293, in load
    return loads(fp.read(),
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
```
Not until I deleted __*settings.json*__ did it go normal.

It seems that the error will be raised if the user didn't complete every setting the last time. Therefore, in case of it, a module that check whether the settings is finished and ask the user to complete it if not should be added.

**Comment by BiDuang** [comment]
Your&nbsp;report&nbsp;has&nbsp;been&nbsp;confirmed.&nbsp;
This&nbsp;bug&nbsp;will&nbsp;be&nbsp;fixed&nbsp;in&nbsp;the&nbsp;next&nbsp;hotfix&nbsp;4.1.
Thanks&nbsp;for&nbsp;your&nbsp;report.
------&nbsp;原始邮件&nbsp;------
***@***.***&gt;;
发送时间:2021年9月6日(星期一)&nbsp;晚上9:11
***@***.***&gt;;
***@***.***&gt;;
主题:[BiDuang/ModSyncer]&nbsp;Fatal:&nbsp;Incomplete&nbsp;*settings.json*&nbsp;Raises&nbsp;Errors&nbsp;the&nbsp;Next&nbsp;Time&nbsp;the&nbsp;Program&nbsp;Runs&nbsp;(#2)



 
I interrupted the program on the fly using ^C (or Ctrl+C in Windows) when testing a new feature, thus leading to an error the next time it ran as below.
 Traceback (most recent call last):   File "~/Desktop/ModSyncer/modsyncer.py", line 59, in <module&gt;     settings = json.load(f)   File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/json/__init__.py", line 293, in load     return loads(fp.read(),   File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/json/__init__.py", line 346, in loads     return _default_decoder.decode(s)  
Not until I deleted settings.json did it go normal.
 
It seems that the error will be raised if the user didn't complete every setting the last time. Therefore, in case of it, a module that check whether the settings is finished and ask the user to complete it if not should be added.
 
&mdash;
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by BiDuang** [comment]
## Your report has been confirmed

Thanks for your report! We have found the cause of this fatal accident. 
In our **guide.py**, an initialization dict will be created and waiting to be written into the **settings.json**. If you terminate the program, the dict will NOT be written to the file and an **EMPTY** file will be created.
This empty json will confuse the `json.load()`, so it will shut the program down and warn you that there was a bad file reading result.

----
This bug will be fixed in next HotFix which version is 4.1. 
Thanks for your support!
